operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -227,20 +339,27 @@ impl<P: Pixel + GdalType> GdalDatasetWriter<P> {
 
         drop(thread_local_configs); // ensure that we drop here
 
-        Ok(Self {
-            dataset,
-            rasterband_index,
-            intermediate_file_path,
-            output_file_path,
-            gdal_tiff_options,
-            gdal_tiff_metadata,
-            output_bounds,
-            output_geo_transform,
-            x_pixel_size,
-            y_pixel_size,
-            _type: Default::default(),
-            use_big_tiff,
-        })
+        Ok(dataset)
+    }
+
+    fn init_new_intermediate_dataset(&mut self) -> Result<()> {
+        self.dataset = Some(Self::create_data_set(
+            &self.gdal_tiff_metadata,
+            self.gdal_tiff_options,
+            &self.gdal_config_options,
+            &self.intermediate_dataset_parameters,
+            self.width,
+            self.height,
+            self.rasterband_index,
+            self.use_big_tiff,
+            self.output_geo_transform,
+        )?);
+
+        Ok(())
+    }
+
+    fn reset_dataset(&mut self) -> Option<Dataset> {
+        std::mem::replace(&mut self.dataset, None)
```

Comment:
Can't you replace that with `self.dataset.take()`?

Reply:
b543a653315680ecef95c34cea64c343fe2f1fc6

Yes.